src/FormLayout/_Abstract.php

```diff
@@ -87,6 +87,37 @@ protected function _addField($decorator, $field)
         return $this->add($decorator, $this->template->hasTag($field->short_name) ? $field->short_name : null);
     }
 
+    /**
+     * Returns array of names of fields to automatically include them in form.
+     *
+     * @param \atk4\data\Model $model
+     *
+     * @return array
+     */
+    protected function getModelFields(\atk4\data\Model $model)
+    {
+        $fields = [];
+        foreach ($model->elements as $f) {
+            if (!$f instanceof \atk4\data\Field) {
+                continue;
+            }
+
+            if ($f->isEditable() || $f->read_only) {
```

Comment:
Automatically include read_only fields in form too if array of field names is not passed as 2nd argument of setModel()

Reply:
This may break some backwards compatibility. For example some read-only fields may be sensitive and shouldn't be displayed.

Reply:
will change to: `if ($f->isEditable() || ($f->read_only && $f->isVisible())`

this also can have some backwards compatibility issues, but it's impossible to fix without that.
Readonly sensitive fields then should be set as not visible `ui`=>['visible'=>false]